bootstrap/src/sun/nio/ch/lincheck/JavaResult.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Lincheck
+ *
+ * Copyright (C) 2019 - 2025 JetBrains s.r.o.
+ *
+ * This Source Code Form is subject to the terms of the
+ * Mozilla Public License, v. 2.0. If a copy of the MPL was not distributed
+ * with this file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package sun.nio.ch.lincheck;
+
+import java.util.concurrent.Callable;
+import java.util.function.Function;
+
+public class JavaResult {
+    private final boolean success;
+    private final Object result;
+    private final Throwable t;
+    
```

Comment:
Let's rename to `InjectedResult` or `BoostrapResult` and use it only in `bootstrap` module, let's use kotlin's `Result` in all other places in the codebase.

Reply:
And consider adding generic parameter `<T>`.